corona-warn-app/cwa-app-android
# Prevent UI scopes from canceling transactions (EXPOSUREAPP-2312)

This introduces a `TransactionScope` that decouples the coroutines/transactions launched in the UI from the UIs `viewScope` as we don't want closed screens to cancel any transactions (coroutines).

Additionally the PR:
* Adds a helper class to further introduce injection to the transactions
* Adds basic tests for transaction behavior
* Removes some unnecessary abstraction and unused code

## How to test
* Execute various transactions, check that they still run through
* Start transactions within the test menu of the `deviceForTesters` flavour, quickly exit the screen and observe logcat, transactions should still finish.
